shared/src/main/scala/mlscript/codegen/Scope.scala

```diff
@@ -57,8 +57,7 @@ class Scope(name: Str, enclosing: Opt[Scope]) {
     this(name, Opt(enclosing))
```

Comment:
Is this constructor still used? If it is, shouldn't the code that uses it be updated too? If yes to both, please do so and remove the constructor.